Source/WebCore/page/Quirks.cpp

```diff
@@ -703,6 +703,21 @@ bool Quirks::shouldAvoidScrollingWhenFocusedContentIsVisible() const
     return needsQuirks() && m_quirksData.shouldAvoidScrollingWhenFocusedContentIsVisibleQuirk;
 }
 
+// Some input only specify image/* as an acceptable type, which is failing sometimes for certains domain names
+// which do not support HEIC.
+bool Quirks::shouldTranscodeHeicImagesForURL(const URL& url)
+{
+    if (!needsQuirks())
```

Comment:
This is a `static` method on the class. It doesn't have any 'state', and doesn't have a Document to ask.

In these cases, we have to check (in the UIProcess side) the page settings `protectedPreferences()->needsSiteSpecificQuirks()` as I showed earlier in this review.